pyresample/bucket/__init__.py

```diff
@@ -296,6 +296,38 @@ def get_max(self, data, fill_value=np.nan, skipna=True):
         LOG.info("Get max of values in each location")
         return self._call_pandas_groupby_statistics('max', data, fill_value, skipna)
 
+    def get_abs_max(self, data, fill_value=np.nan, skipna=True):
+        """Calculate absolute maximums for each bin with drop-in-a-bucket resampling.
+
+        Returns for each bin the original signed value which has the largest
+        absolute value.
+
+        .. warning::
+
+            The slow :meth:`pandas.DataFrame.groupby` method is temporarily used here,
+            as the `dask_groupby <https://github.com/dcherian/dask_groupby>`_ is still under development.
+
+        Parameters
+        ----------
+        data : Numpy or Dask array
+            Data to be binned.
+        skipna : boolean (optional)
+                If True, skips NaN values for the maximum calculation
+                (similarly to Numpy's `nanmax`). Buckets containing only NaN are set to zero.
+                If False, sets the bucket to NaN if one or more NaN values are present in the bucket
+                (similarly to Numpy's `max`).
+                In both cases, empty buckets are set to 0.
+                Default: True
+
+        Returns
+        -------
+        data : Numpy or Dask array
+            Bin-wise maximums in the target grid
+        """
+        max_ = self.get_max(data, fill_value=fill_value, skipna=skipna)
+        min_ = self.get_min(data, fill_value=fill_value, skipna=skipna)
+        return da.where(-min_ > max_, min_, max_)
```

Comment:
I had to think a moment to understand what the `da.where` call is doing. What do you think of moving this to a small method with a self-describing name such as `get_abs_max_from_min_max`? This would also equalize the level of abstraction in this method.

Reply:
Done in 70b24c1.